synapse/config/repository.py

```diff
@@ -180,12 +186,17 @@ def read_config(self, config, **kwargs):
                     e.message  # noqa: B306, DependencyException.message is a property
                 )
 
+            proxies = ["HTTP_PROXY", "HTTPS_PROXY"]
```

Comment:
this feels a bit brittle. It will ignore a lower-case `http_proxy`, for example.

I suggest you import `getproxies_environment` from `urllib.request` and then check the result for `http` and `https`.

Reply:
Ideally we should do whatever we do in the `ProxyAgent`.

Reply:
yup. `ProxyAgent` uses `getproxies_environment`.

Reply:
using `getproxies_environment` seems to be failing the linter with:

```
synapse/config/repository.py:19: error: Module 'urllib.request' has no attribute 'getproxies_environment'
```
Are there any additional changes I need to make to fix this? The other option is to use `getproxies` instead which appears to do the same thing.

Reply:
yeah I don't know why we use `getproxies_environment`. Suggest you just use `getproxies` (but please also update `ProxyAgent` to keep them in sync.)

Reply:
See https://github.com/matrix-org/synapse/pull/9372#issuecomment-777015781 for why we use `getproxies_environment`.

I think we should be consistent (or ideally only call this once and store it on the HS object).

Reply:
@kentokamoto are you still working on this PR?

Reply:
> I think we should be consistent (or ideally only call this once and store it on the HS object).

@clokep we need this information for ProxyAgent and Federation. I am agree with you. In which file / config file should this?

Reply:
Hi @dklimpel Yes I am still working on this. I apologize for taking so long to get this change out. I am keeping my change with `getproxies_environment` as requested for consistency but the linter is failing because it does not recognize the function name. I'm trying to avoid a change where I remove `/synapse/config/repository.py` from the mypy list.

Reply:
Can you push your changes? Then we have a look on it. At the moment I can not see something with `getproxies_environment`.

Reply:
Just pushed the changes. Should be under the hash `3de1c3050a9637c787a4e1868374c95e7024fdce`

Reply:
Ok. I am understand the problem now.
`mypy`  does check `/synapse/config/repository.py` an we get `synapse/config/repository.py:19: error: Module 'urllib.request' has no attribute 'getproxies_environment'`
And `mypy` does **NOT** check `synapse/http/proxyagent.py`. That is the reason why this problem is new.

Reply:
This is a solution
`from urllib.request import getproxies_environment # type: ignore`